packages/core/contracts/oracle/implementation/VotingV2.sol

```diff
@@ -885,6 +891,29 @@ contract VotingV2 is
         batchReveal(revealsAncillary);
     }
 
+    function delegateVoting(address delegate) public {
+        require(voterStakes[delegate].cumulativeStaked == 0, "Cant delegate to existing staker");
+        require(delegateToStaker[delegate] == address(0), "Delegate already used");
+        voterStakes[msg.sender].delegateVoting = delegate;
+    }
+
+    function acceptDelegation(address delegator) public {
+        require(voterStakes[delegator].delegateVoting == msg.sender, "Delegator didn't delegate to you");
+        require(delegateToStaker[msg.sender] == address(0), "Delegation already set");
```

Comment:
If someone proposes a delegate and then the delegate stakes, this could be the case we are trying to avoid right?

I think we should check again here that:
`require(voterStakes[msg.sender].cumulativeStaked == 0, "You cannot accept the delegation if you are staking.");`

Reply:
This is a good call! you are right: you should not be able to accept a delegation if you are currently staked or you can end up in a situation where you cant vote with your staked amount as you are now voting for the account that you delegated to. I've added the require you mentioned.